### ISSUE TITLE
Build broken on LLVM 3.6.2

### ISSUE THREAD

**Issue by kshyatt** [JuliaLang/julia#14450]
```
In file included from /Users/kshyatt/Projects/julia/src/codegen.cpp:708:
./cgutils.cpp:204:25: error: cannot initialize a parameter of type 'llvm::MDNode *' with an lvalue of type 'llvm::DICompileUnit *'
        NMD->addOperand(CU);
                        ^~
/usr/local/Cellar/llvm/3.6.2/include/llvm/IR/Metadata.h:985:27: note: passing argument to parameter 'M' here
  void addOperand(MDNode *M);
                          ^
4 warnings generated.
/Users/kshyatt/Projects/julia/src/codegen.cpp:5369:37: error: use of undeclared identifier 'CU'
            cyclectx->CUs.push_back(CU);
                                    ^
4 warnings generated.
4 warnings and 2 errors generated.
```

@yuyichao can replicate.

**Comment by kshyatt** [comment]
This is also happening to me on `svn` and `3.7.0`.

**Comment by crayxt** [comment]
Can't reproduce on Centos 7 and LLVN_SVN build (make debug release, target is core2)

**Comment by albap** [comment]
I get the same error on LLVM 3.6.2

**Comment by ViralBShah** [comment]
Why aren't you using LLVM 3.7.1?

**Comment by tkelman** [comment]
3.7.1 fails to build with musl libc, which @albap is trying to get working. I'm also interested in that since musl is MIT-licensed and would allow us to avoid some of the hurdles we have to go through involving old Centos versions to make generic Linux binaries.

Might be a small patch to fix LLVM 3.7.1 on musl though. I'll check against the 3.8 release branch too since they're doing RC's right now.

**Comment by vtjnash** [comment]
we'll accept patches, but with 3.8.0 out, we don't officially support versions below 3.7.1
